modules/gdscript/gd_editor.cpp

```diff
@@ -532,6 +532,10 @@ static GDCompletionIdentifier _get_native_class(GDCompletionContext &context) {
 static bool _guess_identifier_type(GDCompletionContext &context, int p_line, const StringName &p_identifier, GDCompletionIdentifier &r_type);
 
 static bool _guess_expression_type(GDCompletionContext &context, const GDParser::Node *p_node, int p_line, GDCompletionIdentifier &r_type) {
+	//if (++context.depth > 100) {
```

Comment:
These comments should not be added.

Instead of reimplementing the changes yourself, you should use `git cherry-pick -x b7fd065f5c067c80a59433ac0a214899ce5bfbac` to properly cherry-pick the original commit.

Reply:
Sorry, in my haste I forgot to remove my working comments.
As for cherry-picking, I am not familiar with it. I tried it and got the warning `warning: inexact rename detection was skipped due to too many files.` and the message `The previous cherry-pick is now empty, possibly due to conflict resolution.`.
I updated the PR with a commit removing them.

Reply:
> `The previous cherry-pick is now empty, possibly due to conflict resolution.`

That's because the cherry-pick would do the same changes as your existing commits, so there was nothing more to apply. What I meant is to do the cherry-pick on the upstream `2.1` branch, superseding your manual redo of the commit.

I've done this and force pushed to update this PR.